### PR TITLE
Feature/det 269 aventri id filter

### PR DESCRIPTION
**Pull request by santoshdasa12345**
## Description of change

New Aventri ID search filter has been added

## Test instructions

New Aventri ID Search Filter has been added
Added Hint text with validations

### Before

image

### After

image

## Checklist

[//]: # 'When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/main/docs/Code%20review%20guidelines.md'

- [x] Has the branch been rebased to main?
- [x] Automated tests (Any of the following when applicable: Unit, Functional or End-to-End)
- [ ] Manual compatibility testing (Browsers: Chrome, Firefox, Edge, Safari)

### CODE REVIEW

File: test/functional/cypress/specs/events/filter-spec.js [hunk 1]
```diff
@@ -558,6 +558,46 @@ describe('events Collections Filter', () => {
...
+        it('should not add anything to the query param if the name is backspaced', () => {
+          cy.get(element).type(`{selectAll}{backspace}{enter}`)
+          cy.url().should('not.include', queryParamWithAventriId)
+        })
+
+        it('should not allow non numerical characters', () => {
+          cy.get(element).type(`${invalidAventriId}{enter}`)
+        })
+
+        it('should allow 9 digits', () => {
```
Comment (chloehkwong1) [inline test/functional/cypress/specs/events/filter-spec.js:584]: Maybe the following test message would be a bit more descriptive? :) 
```suggestion
        it('should truncate any long numbers to 9 digits', () => {
```

File: test/functional/cypress/specs/events/filter-spec.js [hunk 2]
```diff
@@ -558,6 +558,46 @@ describe('events Collections Filter', () => {
         cy.get(latestStartElement).clear()
       })
     })
+
+    context('Aventri ID', () => {
+      const element = '[data-test="aventri-id-filter"]'
+      const aventriId = '200300400'
+      const invalidAventriId = 'Testing'
```
Comment (chloehkwong1) [inline test/functional/cypress/specs/events/filter-spec.js:565]: Maybe add a space and a symbol on the end just to double check it is not allowing those too!
```suggestion
      const invalidAventriId = 'Testing %'
```

File: test/functional/cypress/specs/events/filter-spec.js [hunk 3]
```diff
@@ -558,6 +558,46 @@ describe('events Collections Filter', () => {
...
+      const invalidAventriIdNumbers = '200300400500600'
+      const queryParamWithAventriId = 'aventri_id=200300400'
+
+      context('should filter from user input', () => {
+        it('should aventri ID from user input to query param', () => {
```
Comment (chloehkwong1) [inline test/functional/cypress/specs/events/filter-spec.js:570]:
```suggestion
        it('should add an aventri ID from user input to query param', () => {
```

File: test/functional/cypress/specs/events/filter-spec.js [hunk 4]
```diff
@@ -558,6 +558,46 @@ describe('events Collections Filter', () => {
...
+          cy.url().should('include', queryParamWithAventriId)
+        })
+
+        it('should not add anything to the query param if the name is backspaced', () => {
+          cy.get(element).type(`{selectAll}{backspace}{enter}`)
+          cy.url().should('not.include', queryParamWithAventriId)
+        })
+
+        it('should not allow non numerical characters', () => {
+          cy.get(element).type(`${invalidAventriId}{enter}`)
```
Comment (chloehkwong1) [inline test/functional/cypress/specs/events/filter-spec.js:581]: You should probably check the filter is empty here, and to check nothing as been added to the query params :)

File: test/functional/cypress/specs/events/filter-spec.js [hunk 5]
```diff
@@ -558,6 +558,46 @@ describe('events Collections Filter', () => {
         cy.get(latestStartElement).clear()
       })
     })
+
+    context('Aventri ID', () => {
```
Comment (chloehkwong1) [inline test/functional/cypress/specs/events/filter-spec.js:562]: We should also add in a test to check the express route on the activity feed controller is working correctly (let me know when you get to this stage and I can help!)

**Review by chloehkwong1 — CHANGES_REQUESTED** [review]

File: test/functional/cypress/specs/events/filter-spec.js (hunk 4 above)
Comment (santoshdasa12345) [reply test/functional/cypress/specs/events/filter-spec.js:581]: Added assertion for the test

File: test/functional/cypress/specs/events/filter-spec.js (hunk 5 above)
Comment (santoshdasa12345) [reply test/functional/cypress/specs/events/filter-spec.js:575]: Add tests

**Review by richmadetech — APPROVED** [review]

**Review by onyela-o — APPROVED** [review]

**Review by chloehkwong1 — APPROVED** [review]

**Pull request merged**